File: go/src/maxProfit/maxProfit.go
package maxProfit

//https://leetcode-cn.com/problems/best-time-to-buy-and-sell-stock-ii/solution/mai-mai-gu-piao-de-zui-jia-shi-ji-ii-by-leetcode-s/
func maxProfit(prices []int) int {
	var ans, n = 0, len(prices)

	for i := 1; i < n; i++ {
		ans += max(0, prices[i] - prices[i - 1])
	}

	return ans
}

func max(x, y int) int {
	if x > y {
		return x
	}

	return y
}

File: c/src/maxProfit/maxProfit.c
#include "maxProfit.h"

int maxProfit(int* prices, int pricesSize){
    int i = 1, ans = 0;

    for (;i < pricesSize; ++i) {
        if (prices[i] > prices[i - 1])
            ans += prices[i] - prices[i - 1];
    }

    return ans;
}